### PR TITLE
Person photo refactoring

### PR DESCRIPTION
**Pull request by Space-tourist**

### CODE REVIEW

File: src/Inc.TeamAssistant.Connector.Application/QueryHandlers/GetPersonPhoto/GetPersonPhotoQueryHandler.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+internal sealed class GetPersonPhotoQueryHandler : IRequestHandler<GetPersonPhotoQuery, GetPersonPhotoResult>
+{
+    private readonly PersonPhotosService _personPhotosService;
+
+    public GetPersonPhotoQueryHandler(PersonPhotosService personPhotosService)
+    {
+        _personPhotosService = personPhotosService ?? throw new ArgumentNullException(nameof(personPhotosService));
+    }
+
+    public async Task<GetPersonPhotoResult> Handle(GetPersonPhotoQuery query, CancellationToken cancellationToken)
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Connector.Application/QueryHandlers/GetPersonPhoto/GetPersonPhotoQueryHandler.cs:16]: cancellationToken -> token

File: src/Inc.TeamAssistant.Connector.DataAccess/PersonRepository.cs [hunk 2]
```diff
@@ -138,4 +138,23 @@ UPDATE connector.teammates
...
+    public async Task<Guid> GetBotId(
+        long personId,
+        CancellationToken token)
+    {
+        var command = new CommandDefinition(@"
+            SELECT
+                t.bot_id AS botid
+            FROM connector.teammates AS tm
+            JOIN connector.teams AS t ON tm.team_id = t.id
+            WHERE tm.person_id = @person_id;",
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Connector.DataAccess/PersonRepository.cs:151]: Person can has many bots. Please add add limit 1 to this query

File: src/Inc.TeamAssistant.Connector.Model/Queries/GetPersonPhoto/PhotoDto.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+namespace Inc.TeamAssistant.Connector.Model.Queries.GetPersonPhoto;
+
+public sealed record PhotoDto(
+    Guid Id,
+    long PersonId,
+    DateTimeOffset Date,
+    byte[] Data);
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Connector.Model/Queries/GetPersonPhoto/PhotoDto.cs:7]: You use only date field from this DTO. Please remove this record.

File: src/Inc.TeamAssistant.Connector.Application/Services/PersonPhotosService.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        ILogger<PersonPhotosService> logger,
+        IPersonRepository personRepository)
+    {
+        _telegramBotClientProvider = telegramBotClientProvider ??
+                                     throw new ArgumentNullException(nameof(telegramBotClientProvider));
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _personRepository = personRepository ?? throw new ArgumentNullException(nameof(personRepository));
+    }
+
+    public async Task<Photo?> GetPersonPhoto(long personId, CancellationToken token)
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Connector.Application/Services/PersonPhotosService.cs:25]: You can return 'stream?' from this method. Don't remember set stream position to 0 and remove 'using'.

File: src/Inc.TeamAssistant.Migrations/2024_07_27_0_DeletePhotos.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+{
+    public override void Up()
+    {
+        Delete
+            .Table("photos")
+            .InSchema("connector");
+    }
+
+    public override void Down()
+    {
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Migrations/2024_07_27_0_DeletePhotos.cs:16]: Please write code for rollback changes. Create empty table

File: src/Inc.TeamAssistant.Connector.Application/QueryHandlers/GetPersonPhoto/GetPersonPhotoQueryHandler.cs (hunk 1 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Connector.Application/QueryHandlers/GetPersonPhoto/GetPersonPhotoQueryHandler.cs]: fixed

File: src/Inc.TeamAssistant.Connector.DataAccess/PersonRepository.cs (hunk 2 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Connector.DataAccess/PersonRepository.cs]: fixed

File: src/Inc.TeamAssistant.Connector.Model/Queries/GetPersonPhoto/PhotoDto.cs (hunk 3 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Connector.Model/Queries/GetPersonPhoto/PhotoDto.cs]: fixed

File: src/Inc.TeamAssistant.Connector.Application/Services/PersonPhotosService.cs (hunk 4 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Connector.Application/Services/PersonPhotosService.cs]: fixed

File: src/Inc.TeamAssistant.Migrations/2024_07_27_0_DeletePhotos.cs (hunk 5 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Migrations/2024_07_27_0_DeletePhotos.cs:16]: fixed

File: src/Inc.TeamAssistant.Connector.Application/QueryHandlers/GetPersonPhoto/GetPersonPhotoQueryHandler.cs [hunk 6]
```diff
@@ -13,13 +13,14 @@ public GetPersonPhotoQueryHandler(PersonPhotosService personPhotosService)
...
-    public async Task<GetPersonPhotoResult> Handle(GetPersonPhotoQuery query, CancellationToken cancellationToken)
+    public async Task<GetPersonPhotoResult> Handle(GetPersonPhotoQuery query, CancellationToken token)
     {
         ArgumentNullException.ThrowIfNull(query);
+
+        using var stream = await _personPhotosService.GetPersonPhoto(query.PersonId, token);
         
-        var photo = await _personPhotosService.GetPersonPhoto(query.PersonId, cancellationToken);
-        var photoDto = photo is null ? null : new PhotoDto(photo.Id, photo.PersonId, photo.Date, photo.Data);
+        var photoData = stream is null ? Array.Empty<byte>() : stream.ToArray();
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Connector.Application/QueryHandlers/GetPersonPhoto/GetPersonPhotoQueryHandler.cs:22]: Please do not get byte array from stream. Return stream to controller and create action result from stream.

File: src/Inc.TeamAssistant.Gateway/Controllers/PhotosController.cs [hunk 7]
```diff
@@ -24,8 +24,8 @@ public async Task<IActionResult> Get(long personId, CancellationToken token)
         const string contentType = "image/jpeg";
         var result = await _mediator.Send(new GetPersonPhotoQuery(personId), token);
         
-        return result.Photo is null
+        return !result.PhotoData.Any()
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Gateway/Controllers/PhotosController.cs:27]: Please invert condition for more readable code style. 'result.PhotoData.Any() ? File(result.PhotoData, contentType) : File("imgs/user_stub.jpg", contentType)'

**Review by dyatlov-a — APPROVED** [review]

**Pull request merged**